bundles/org.openhab.binding.ring/src/main/java/org/openhab/binding/ring/internal/data/RingDevices.java

```diff
@@ -0,0 +1,172 @@
+     * "motion_snooze_presets": [
+     * "none",
+     * "low",
+     * "medium",
+     * "high"
+     * ],
+     * "live_view_preset_profile": "middle",
+     * "live_view_presets": [
+     * "low",
+     * "middle",
+     * "high",
+     * "highest"
+     * ],
+     * "pir_sensitivity_1": 0,
+     * "vod_suspended": 0,
+     * "doorbell_volume": 1,
+     * "vod_status": "enabled"
+     * },
+     * "features": {
+     * "motions_enabled": true,
+     * "show_recordings": false,
+     * "show_vod_settings": true
+     * },
+     * "owned": true,
+     * "alerts": {
+     * "connection": "online"
+     * },
+     * "motion_snooze": null,
+     * "stolen": false,
+     * "location_id": null,
+     * "ring_id": null,
+     * "owner": {
+     * "id": 4445516,
+     * "first_name": null,
+     * "last_name": null,
+     * "email": "xxx@acme.com"
+     * }
+     * }
+     * ],
+     * "authorized_doorbots": [],
+     * "chimes": [],
+     * "stickup_cams": [],
+     * "base_stations": []
+     * } *
+     *
+     * @param jsonRingDevices the JSON ring devices retrieved from the Ring API.
+     */
+    public RingDevices(JSONObject jsonRingDevices, RingAccount ringAccount) {
+        addDoorbells((JSONArray) jsonRingDevices.get(ApiConstants.DEVICES_DOORBOTS), ringAccount);
+        addStickupCams((JSONArray) jsonRingDevices.get(ApiConstants.DEVICES_STICKUP_CAMS), ringAccount);
+    }
+
+    /**
+     * Helper method to create the doorbell list.
+     *
+     * @param jsonDoorbells
+     */
+    private final void addDoorbells(JSONArray jsonDoorbells, RingAccount ringAccount) {
+        doorbells = new ArrayList<>();
+        for (Object obj : jsonDoorbells) {
+            Doorbell doorbell = new Doorbell((JSONObject) obj);
+            doorbell.setRingAccount(ringAccount);
+            doorbells.add(doorbell);
+        }
+    }
+
+    /**
+     * Retrieve the Doorbells Collection.
+     *
+     * @return
+     */
+    public Collection<Doorbell> getDoorbells() {
+        return doorbells;
+    }
+
+    /**
+     * Retrieve the Stickupcams Collection.
+     *
+     * @return
+     */
+    public Collection<Stickupcam> getStickupcams() {
+        return stickupcams;
+    }
+
+    /**
+     * Helper method to create the stickupcam list.
+     *
+     * @param jsonStickupcams
+     */
+    private final void addStickupCams(JSONArray jsonStickupcams, RingAccount ringAccount) {
+        stickupcams = new ArrayList<>();
+        for (Object obj : jsonStickupcams) {
+            Stickupcam stickupcam = new Stickupcam((JSONObject) obj);
+            stickupcam.setRingAccount(ringAccount);
+            stickupcams.add(stickupcam);
+        }
+    }
+
+    /**
+     * Retrieve a collection of all devices.
+     *
+     * @return
+     */
+    public Collection<RingDevice> getRingDevices() {
+        List<RingDevice> result = new ArrayList<>();
+        result.addAll(doorbells);
+        result.addAll(stickupcams);
+        return result;
```

Comment:
replace by return `List.of(doorbells,stickupcams);`

Reply:
[ERROR] /home/openhab/ring/openhab-addons/bundles/org.openhab.binding.ring/src/main/java/org/openhab/binding/ring/internal/data/RingDevices.java:[167,35] Type mismatch: cannot convert from java.util.List<java.util.List<? extends org.openhab.binding.ring.internal.data.AbstractRingDevice>> to java.util.List<org.openhab.binding.ring.internal.data.RingDevice>